x-element.js

```diff
@@ -447,7 +447,7 @@ export default class XElement extends HTMLElement {
   // Called once per-property during constructor analysis.
   static __mutateProperty(constructor, propertyMap, key, property) {
     property.key = key;
-    property.attribute = !property.internal && XElement.__serializableTypes.has(property.type)
+    property.attribute = !property.internal && (XElement.__serializableTypes.has(property.type) || !property.type)
```

Comment:
This assignment ternary is becoming difficult to follow, can we unpack it a bit?

Reply:
Sure, I'll take a look at it 👌